prov/verbs/src/verbs_info.c

```diff
@@ -641,8 +641,8 @@ static int fi_ibv_get_device_attrs(struct ibv_context *ctx, struct fi_info *info
 
 	info->domain_attr->cq_cnt 		= device_attr.max_cq;
 	info->domain_attr->ep_cnt 		= device_attr.max_qp;
-	info->domain_attr->tx_ctx_cnt 		= MIN(info->domain_attr->tx_ctx_cnt, device_attr.max_qp);
-	info->domain_attr->rx_ctx_cnt 		= MIN(info->domain_attr->rx_ctx_cnt, device_attr.max_qp);
+	info->domain_attr->tx_ctx_cnt 		= MIN(info->domain_attr->tx_ctx_cnt, device_attr.max_qp_wr);
+	info->domain_attr->rx_ctx_cnt 		= MIN(info->domain_attr->rx_ctx_cnt, device_attr.max_qp_wr);
```

Comment:
This should remain device_attr.max_qp as tx/rx_ctx_cnt corresponds to the # of QPs. The max # of WRs that can be posted to a send/recv queue can be got from info->tx_attr/rx_attr->size respectively. The function fi_ibv_get_qp_cap() initializes these values.